test/test_functional.py

```diff
@@ -0,0 +1,182 @@
+import math
+import os
+
+import torch
+import torchaudio
+import unittest
+import test.common_utils
+
+
+class TestFunctional(unittest.TestCase):
+    data_sizes = (2, 20)
+    number_of_trials = 100
+
+    def _compare_estimate(self, sound, estimate, atol=1e-6, rtol=1e-8):
+        # trim sound for case when constructed signal is shorter than original
+        sound = sound[..., :estimate.size(-1)]
+
+        self.assertTrue(sound.shape == estimate.shape, (sound.shape, estimate.shape))
+        self.assertTrue(torch.allclose(sound, estimate, atol=atol, rtol=rtol))
+
+    def _test_istft_is_inverse_of_stft(self, kwargs):
+        # generates a random sound signal for each tril and then does the stft/istft
+        # operation to check whether we can reconstruct signal
+        for i in range(self.number_of_trials):
+            sound = torch.rand(self.data_sizes)
```

Comment:
this should be deterministic. If the test fails you don't have the data to reproduce it. So then you'll start creating all kinds of data to see if you can get a deterministic reproduction. you can write out some simple pseudo-random number generator that takes a seed.

Reply:
that'd be useful in general for other case s as well